application/cache/cache.go

```diff
@@ -0,0 +1,233 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/go-redis/cache/v8"
+	"github.com/go-redis/redis/v8"
+
+	"github.com/silinternational/wecarry-api/api"
+	"github.com/silinternational/wecarry-api/conversions"
+	"github.com/silinternational/wecarry-api/models"
+)
+
+var (
+	RequestsCache         *cache.Cache
+	PublicRequestsKeyName string = "public"
+)
+
+func init() {
+	ring := redis.NewRing(&redis.RingOptions{
+		Addrs: map[string]string{
+			"redis": "redis:6379",
+		},
+	})
+
+	RequestsCache = cache.New(&cache.Options{
+		Redis:      ring,
+		LocalCache: cache.NewTinyLFU(1000, time.Minute),
+		// use json in lieu of msgpack for encoding/decoding
+		Marshal:   json.Marshal,
+		Unmarshal: json.Unmarshal,
+	})
+}
+
+// Wrapper for cache write
+func CacheWrite(ctx context.Context, organization string, requestsMap interface{}) error {
```

Comment:
This method name is generic, but the operation details are specific to `requests`. We should either make the method name requests specific as well, or allow the method to be more generic. I assume we'll cache other kinds of things over time, so not having to refactor this later would be nice. Maybe having thin wrapper/helper methods for different types so that we can implement the cache key generation rules there, then call a shared Set method like you have below. For requests I'd expect keys like `requests-orgname-only`, `requests-orgname-affiliated`, `requests-allusers`, etc.